chall01/asolopov.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python3
+import sys
+
+morse_dict = {
+"a": ".-",
+"b": "-...",
+"c": "-.-.",
+"d": "-..",
+"e": ".",
+"f": "..-.",
+"g": "--.",
+"h": "....",
+"i": "..",
+"j": ".---",
+"k": "-.-",
+"l": ".-..",
+"m": "--",
+"n": "-.",
+"o": "---",
+"p": ".--.",
+"q": "--.-",
+"r": ".-.",
+"s": "...",
+"t": "-",
+"u": "..-",
+"v": "...-",
+"w": ".--",
+"x": "-..-",
+"y": "-.--",
+"z": "--.."
+}
+
+def usage():
+	print(f"usage: {sys.argv[0]} <a-zA-Z string>")
+	exit()	
+
+def check_chars(word, morse_dict):
+	for x in word:
+		if x.isalpha() == False:
+			if x.isspace() == False:
+				usage()
+		else:
+			if x not in morse_dict:
+				usage()
+
+if (len(sys.argv)) != 2 or not sys.argv[1]:
+	usage()
+else:
+	word = sys.argv[1].lower()
+	check_chars(word, morse_dict)
+	for x in word:
+		if (x.isalpha()):
```

Comment:
One suggestion is to add the space key/value pair in your morse_dict. That would allow you to remove that check of isalpha or isspace. Same as in your `check_chars()` function, it would prevent you this different checks, you could then just do 
```
if x not in morse_dict:
  usage()
```